quinn-proto/src/connection.rs

```diff
@@ -587,22 +677,25 @@ impl Connection {
         &mut self,
         mux: &mut impl Multiplexer,
         now: u64,
+        ecn: Option<EcnCodepoint>,
         packet: Option<u64>,
     ) {
         self.reset_idle_timeout(mux, now);
-        let packet = if let Some(x) = packet {
-            x
-        } else {
-            return;
-        };
-        trace!(self.log, "packet authenticated"; "pn" => packet);
-        self.pending_acks.insert_one(packet);
-        if self.pending_acks.len() > MAX_ACK_BLOCKS {
-            self.pending_acks.pop_min();
+        self.receiving_ecn |= ecn.is_some();
+        if let Some(x) = ecn {
+            self.ecn_counters += x;
         }
-        if packet > self.rx_packet {
-            self.rx_packet = packet;
-            self.rx_packet_time = now;
+
+        if let Some(packet) = packet {
```

Comment:
I think I prefer the previous style of returning early for `None` `packet`s. Also, this is another entirely non-functional change intermixed in a commit that otherwise very much isn't.